components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -262,6 +262,13 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
                     }
                 }
 
+                ///Check the requirement of having a configured public certificate or JWKS in SP
+                boolean isIdTokenEncryptionEnabled = application.isIdTokenEncryptionEnabled();
+                boolean isRequestObjectSignatureValidationEnabled =
+                        application.isRequestObjectSignatureValidationEnabled();
+                handlePublicCertificateConfig(app.getOauthConsumerKey(), tenantDomain, isIdTokenEncryptionEnabled,
```

Comment:
Shouldn't we call this method only if either isIdTokenEncryptionEnabled or isRequestObjectSignatureValidationEnabled is true

Reply:
addressed with cc43b59fd26aeaf1931e9600fd8ac852fe1fd31d